cg/models/nf_analysis.py

```diff
@@ -44,22 +42,24 @@ def fastq_files_exist(cls, fastq_paths: List[str], values: dict) -> List[str]:
         return fastq_paths
 
 
-class NextflowDeliverables(BaseModel):
-    """Nextflow deliverables model
+class FileDeliverable(BaseModel):
+    """Specification for a general deliverables file."""
 
-    Attributes:
-        deliverables: dictionary containing format, path, path_index, step, tag and id keys.
-    """
+    id: str
+    format: str
+    path: FilePath
+    path_index: Optional[FilePath]
+    step: str
+    tag: str
+
+    @validator("path", "path_index")
+    def convert_path_to_string(cls, file_path):
+        if file_path and isinstance(file_path, Path):
+            return str(file_path)
```

Comment:
- FilePath vs Path: FilePath checks that the file exists while Path only requires the passed object to be of the class Path.
- FilePath does not want a string, wants a Path. I am the one who wants to convert it to a string after making sure that the file exists. This is to avoid writing issues with when writing the output. But I do agree that is counterintuitive. I wasn't sure how to make it more intuitive. I happily take suggestions

Reply:
Keep it simple - I would just use Path which a standard module and well known to most. Make a very general Pydantic validator that can be reused by many models - place it perhaps under `apps/pydantic/validators.py` or similar @islean (preference?).
Use pathlibs `is_file` to perform the check.

Reply:
@henrikstranneheim @fevac Yeah, maybe write the validator as generally as possible, then place it as narrow as you can? E.g. if it just joining a List[str] to a string, then it can go into apps/pydantic/validators, but if it is a field validator only used for one particular model, either create a "validators" file in the model directory or create a sub directory to apps/pydantic/validators/<relevant placement>?

Reply:
I think moving validators out of the BaseModel is only supported in pydantic v2, though.

Reply:
The validators should work no matter which library we use, maybe we can just call it validators an omit the pydantic part

Reply:
I've updated the type hints so they are not misleading and updated the validator as pre=True. Other changes seem to be related pydantic v2 but I would rather apply those in a separate pr